50_wa_diff_overdose.py

```diff
@@ -0,0 +1,299 @@
+        return 1
+
+    else:
+        return "error"
+
+
+# copy to fix the dreaded "A value is trying to be set on a copy of a slice" error
+wa_deaths = washington.copy()
+comp_deaths = comp.copy()
+
+# apply new function to our df
+wa_deaths["overdose"] = wa_deaths["Drug/Alcohol Induced Cause"].apply(
+    lambda x: select_overdose(x)
+)
+comp_deaths["overdose"] = comp_deaths["Drug/Alcohol Induced Cause"].apply(
+    lambda x: select_overdose(x)
+)
+
+# filter accordingly based on new column
+wa_deaths = wa_deaths[wa_deaths["overdose"] != 0]
+comp_deaths = comp_deaths[comp_deaths["overdose"] != 0]
+
+wa_deaths["overdose_per_100k"] = wa_deaths["Deaths"] / wa_deaths["Population"] * 100_000
+comp_deaths["overdose_per_100k"] = (
+    comp_deaths["Deaths"] / comp_deaths["Population"] * 100_000
+)
+
+wa_result = (
+    wa_deaths.groupby(["Year", "County"])["overdose_per_100k"].sum().reset_index()
+)
+comp_result = (
+    comp_deaths.groupby(["Year", "StateName", "CountyName"])["overdose_per_100k"]
+    .sum()
+    .reset_index()
+)
+comp_result[comp_result["StateName"] == "Oregon"].groupby("Year")[
+    "overdose_per_100k"
+].mean()
+comp_result[comp_result["StateName"] == "Idaho"].groupby("Year")[
+    "overdose_per_100k"
+].mean()
+comp_result[comp_result["StateName"] == "Montana"].groupby("Year")[
+    "overdose_per_100k"
+].mean()
+comp_result[comp_result["StateName"] == "Nevada"].groupby("Year")[
+    "overdose_per_100k"
+].mean()
+comp_result[comp_result["StateName"] == "Wyoming"].groupby("Year")[
+    "overdose_per_100k"
+].mean()
+wa_result.groupby("Year")["overdose_per_100k"].mean()
+comp_result.groupby("Year")["overdose_per_100k"].mean()
+wa_result = wa_result.groupby("Year")["overdose_per_100k"].mean().reset_index()
+
+comp_result = comp_result.groupby(["Year"])["overdose_per_100k"].mean().reset_index()
+# create a scale for number of years before and after 2012 (target year)
+
+
+def scale_years(year):
+    if year == 2009:
+        return -3
+    if year == 2010:
+        return -2
+    if year == 2011:
+        return -1
+    if year == 2012:
+        return 0
+    if year == 2013:
+        return 1
+    if year == 2014:
+        return 2
+    if year == 2015:
+        return 3
+
+
+wa_result["year relative to policy"] = wa_result["Year"].apply(lambda x: scale_years(x))
+comp_result["year relative to policy"] = comp_result["Year"].apply(
+    lambda x: scale_years(x)
+)
+
+# double check no nulls in "year relative to policy"
+
+assert wa_result["year relative to policy"].isnull().sum() == 0
+assert comp_result["year relative to policy"].isnull().sum() == 0
+
+# doing this in case the float == int comparison causes issues
+# split into before 2007 and after 2007
+
+wa_b4 = wa_result[wa_result["Year"] < 2012]
+wa_after = wa_result[wa_result["Year"] >= 2012]
+
+
+wa_after = wa_after[wa_after["Year"] != 2012]  # may need to handle this differently
+
+from sklearn.linear_model import LinearRegression
+
+regressor_b4 = LinearRegression()
```

Comment:
Again as suggested for line 134, I think the name regressor_before is more clear and understandable. Looks a bit more formal compared to regressor_b4.